kernel/src/main/scala/cats/kernel/instances/SortedSetInstances.scala

```diff
@@ -23,12 +23,20 @@ private[instances] trait SortedSetInstances1 {
 class SortedSetOrder[A: Order] extends Order[SortedSet[A]] {
   def compare(a1: SortedSet[A], a2: SortedSet[A]): Int =
     cats.kernel.instances.int.catsKernelStdOrderForInt.compare(a1.size, a2.size) match {
-      case 0 => StaticMethods.iteratorCompare(a1.iterator, a2.iterator)
+      case 0 =>
+        // In the event that the ordering instances are not _exactly_ the
+        // same, we have to sort/rebuild the sets. If we don't we risk
+        // violating the Monoid identity law.
+        if (a1.ordering eq a2.ordering) {
```

Comment:
`eq` is too restrictive - `Ordering` defines structural equality for the builtin cases
```suggestion
        if (a1.ordering == a2.ordering) {
```

Reply:
:+1: I'll update that.

Reply:
Scala 2.13 also defines `Ordering.isReverseOf` so if we're willing to use version-specific sources we could add another special case to reverse one of the sequences (or e.g. do `toIndexedSeq.reverseIterator`) instead of sorting both 😄

Reply:
Wait, this compares `Ordering[A]` _instances_... I think `eq` makes perfect sense.

Reply:
`eq` works only for primitive types but not for `List`, `Option`, tuple, etc. - what's wrong with using `==`?

Reply:
@durban `==` may still yield false for instances which are logically equivalent, e.g. `new Ordering[A]{ override def compare(a: A, b: B): Int = Ordering[A].compare(a, b)}`, but that's okay. The danger would be if it yielded `true` for logically equivalent instances.

As @joroKr21 points out, for _some_ cases we can use structural equality from the stdlib to avoid the expensive branch, e.g. https://github.com/scala/scala/blob/v2.13.6/src/library/scala/math/Ordering.scala#L259

Reply:
There is nothing wrong with `==`; I just wasn't aware that there are `Ordering` instances which use non-reference equality. Apparently they exist, so `==` makes even more sense than `eq` :-)

(Btw. it is possible, for `==` to return true for logically not equivalent `Ordering` instances, but that is a bug in the `Ordering` instance.)

Reply:
Basically the standard library had the same problem and came up with this solution.
Ironically `Ordering` instances created from `Order` are neither `==` nor `eq` 😞

Reply:
We could fix our `Ordering` from `Order` somewhat, but yeah, cats has not tried to make equality work for typeclass values.

BTW: having written several compilers and compiler-like things in scala, making equivalent things equal can be very useful for term rewriting that often happens there (i.e. taking some expressions written in scala and seeing if they are equal to others).